kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/JMXQuartzSchedulerTest.java

```diff
@@ -0,0 +1,221 @@
+@DirtiesContext(classMode = ClassMode.AFTER_CLASS)
+public class JMXQuartzSchedulerTest {
+
+	private static final String GROUP_ID = "org.jbpm";
+	private static final String ARTIFACT_ID = "intermediate-timer-sample";
+	private static final String VERSION = "1.0";
+
+	private static final String PROCESS_ID = "org.jbpm.sample.intermediate-timer";
+
+	private KModuleDeploymentUnit unit = null;
+	private MBeanServer mBeanServer = null;
+	private ObjectName oName = null;
+
+	@Autowired
+	private ProcessService processService;
+
+	@Autowired
+	private DeploymentService deploymentService;
+
+	@Autowired
+	private CountDownLatchEventListener countDownListener;
+
+	@BeforeClass
+	public static void generalSetup() {
+		KieServices ks = KieServices.Factory.get();
+		org.kie.api.builder.ReleaseId releaseId = ks.newReleaseId(GROUP_ID, ARTIFACT_ID, VERSION);
+		File kjar = new File("../kjars/intermediate-timer-sample/intermediate-timer-sample-1.0.jar");
+		File pom = new File("../kjars/intermediate-timer-sample/pom.xml");
+		MavenRepository repository = getMavenRepository();
+		repository.installArtifact(releaseId, kjar, pom);
+	}
+
+	@Before
+	public void setup() {
+		unit = new KModuleDeploymentUnit(GROUP_ID, ARTIFACT_ID, VERSION);
+		deploymentService.deploy(unit);
+	}
+
+	@After
+	public void cleanup() {
+		deploymentService.undeploy(unit);
+	}
+
+	@Test(timeout = 30000)
+	public void whenSchedulerStartedThenTimerIsFired() throws Exception {
+		countDownListener.configure(PROCESS_ID, 1);
+
+		long processInstanceId = processService.startProcess(unit.getIdentifier(), PROCESS_ID);
+		assertNotNull(processInstanceId);
+
+		countDownListener.getCountDown().await();
+		assertTrue(countDownListener.getExecutingThread().startsWith("SpringBootScheduler"));
+
+		ProcessInstance pi = processService.getProcessInstance(processInstanceId);
+		assertNull(pi);
+	}
+
+	@Test(timeout = 30000)
+	public void whenSchedulerPausedThenTimerNotFired() throws Exception {
+		long processInstanceId = processService.startProcess(unit.getIdentifier(), PROCESS_ID);
+		assertNotNull(processInstanceId);
+
+		pauseScheduler();
+
+		// Wait 20 seconds, as the timer was scheduled for 15 seconds
+		CountDownLatch latch = new CountDownLatch(1);
+		latch.await(20, TimeUnit.SECONDS);
+
+		ProcessInstance pi = processService.getProcessInstance(processInstanceId);
+		assertNotNull(processInstanceId);
+
+		assertEquals(STATE_ACTIVE, pi.getState());
+
+		processService.abortProcessInstance(processInstanceId);
+	}
+
+	@Test(timeout = 40000)
+	public void whenSchedulerRestartedThenTimerIsFired() throws Exception {
+		countDownListener.configure(PROCESS_ID, 1);
+
+		long processInstanceId = processService.startProcess(unit.getIdentifier(), PROCESS_ID);
+		assertNotNull(processInstanceId);
+
+		pauseScheduler();
+		
+		// Wait 20 seconds, as the timer was scheduled for 15 seconds
+		CountDownLatch latch = new CountDownLatch(1);
+		latch.await(20, TimeUnit.SECONDS);
```

Comment:
Same here. And later you can again call await on the same countDownListener.

Reply:
OK, same resolution, thanks!